### ISSUE TITLE
redundant_closure calls a tuple struct enum variant a function

### ISSUE THREAD

**Issue by skrapi** [rust-lang/rust-clippy#8817]
### Summary

Basically Clippy lint calls the tuple struct that I am using a function which is incorrect, and led to me being confused about what it wanted me to do. 

### Reproducer

I tried this code:

```rust
use std::collections::HashSet;
use std::collections::TryReserveError;

#[derive(Debug)]
enum MyError {
    A(TryReserveError)
}


fn main() {
    let mut hashset: HashSet<u32> = HashSet::<u32>::new();

    hashset.try_reserve(10).map_err(|error| MyError::A(error)).unwrap();
}

```

I expected to see this happen:
Clippy should tell me that I can just call the tuple struct

Instead, this happened:
Clippy calls the tuple struct a function, which confused me.

```sh
Checking playground v0.0.1 (/playground)
warning: redundant closure
  [--> src/main.rs:13:37
](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=e6487c80374671aecf918e801f279c42#)   |
13 |     hashset.try_reserve(10).map_err(|error| MyError::A(error)).unwrap();
   |                                     ^^^^^^^^^^^^^^^^^^^^^^^^^ help: replace the closure with the function itself: `MyError::A`
   |
   = note: `#[warn(clippy::redundant_closure)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#redundant_closure

warning: `playground` (bin "playground") generated 1 warning
    Finished dev [unoptimized + debuginfo] target(s) in 0.44s
    ```

**Comment by PatchMixolydic** [comment]
Technically, tuple struct constructors are functions. In this example, `MyError::A: fn(TryReserveError) -> MyError` (ignoring unique function types). However, this diagnostic (and others) should almost definitely say "tuple struct constructor" in this case to reduce confusion.
